stable_baselines3/common/vec_env/dummy_vec_env.py

```diff
@@ -51,6 +51,8 @@ def step_wait(self) -> VecEnvStepReturn:
         return (self._obs_from_buf(), np.copy(self.buf_rews), np.copy(self.buf_dones), deepcopy(self.buf_infos))
 
     def seed(self, seed: Optional[int] = None) -> List[Union[None, int]]:
+        if seed is None:
```

Comment:
why not just return in that case?

Reply:
From intuition I would assume that if I call `env.seed()` without parameters, that a seed would be chosen for me randomly.
I mostly want to ensure, that all envs of the vecenv have a different seed. If we just return, all of them will have the same seed, when the processes are forked.

Reply:
I see your point, I thought that the seed would have been different per process but a quick test shows it's not the case when processes are forked (which is not the default method of SB3 luckily), so the following yield a different result per process:

```python
import multiprocessing as mp

import numpy as np

def f(x):
    return np.random.rand()

if __name__ == '__main__':
    ctx = mp.get_context('spawn')
    with ctx.Pool(3) as p:
        print(p.map(f, [1, 1, 1]))
```

Reply:
Huh, for me the output of your example is the same over all processes with fork.
```
import multiprocessing as mp

import numpy as np

def f(x):
    return np.random.rand()

if __name__ == '__main__':
    ctx = mp.get_context('fork')
    with ctx.Pool(3) as p:
        print(p.map(f, [1, 1, 1]))
```
prints
```
[0.5667669684783624, 0.5667669684783624, 0.5667669684783624]
```

Yeah, sure, the `seed = np.random.randint(0, 2**32 - 1)` will be deterministic, if you force a seed before, but the subprocesses will still have their own unique seed.

Reply:
> Huh, for me the output of your example is the same over all processes with fork.

yes, but that's not the case for `SuprocVecEnv` with start method fork for some reasons... (see test below, you can comment out `seed = np.random.randint(0, 2**32 - 1)` part and return None to check what happens)

Reply:
> with start method fork for some reasons...

nevermind, I think the example in the test is using a separate random seed generator. Let me try.

Reply:
See https://github.com/DLR-RM/stable-baselines3/pull/808/commits/a64714b546b82841bed456c53c36c6e84fec6392